### PR TITLE
[feat]: Create okr objective component

### PR DESCRIPTION
**Pull request by Nasaa1234**
- [x] Implemented Create Okr objective component.
- [x] Objective Edit, Delete, Add 
- [x] Add sub Objective
path: http://localhost:3000/okr/MainScree
Screen Shot 2022-08-05 at 17 00 50
n

### CODE REVIEW

**Review by M4nduul — CHANGES_REQUESTED** [review]
Edit delete hiihed asuudaltai bn

**Comment by Nasaa1234** [comment]
> Edit delete hiihed asuudaltai bn

code ni uu?

File: frontend/src/components/okr/mainsSreen/Accordion.tsx [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+  edit: {
+    display: "flex",
+    flexDirection: "row",
+    alignItems: "center",
+    justifyContent: "center",
+  },
+};
+
+const OkrAccordion: FC<PropsType> = props => {
+  const { title, data, setData, ind } = props;
```
Comment (sumiyab) [inline frontend/src/components/okr/mainsSreen/Accordion.tsx:50]: nerlehdee descriptive name ugmuur bn

File: frontend/src/components/okr/mainsSreen/Accordion.tsx [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+                  index={index}
+                  setCollection={setCollection}
+                  edit={edit}
+                  collection={collection}
+                  key={index}
+                  setEdit={setEdit}
+                />
+              );
+            })}
+
```
Comment (sumiyab) [inline frontend/src/components/okr/mainsSreen/Accordion.tsx:126]: descriptive name

File: frontend/src/components/okr/mainsSreen/More.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      setType("points");
+      setOpen(true);
+    }
+    if (title === "Delete to") {
+      setData(data.filter((_el, index) => index !== ind1));
+    }
+    if (title === "Edit Result") {
+      setEdit({ result: true, goals: false, resultIndex: ind });
+    }
+    if (title === "Delete Result") {
```
Comment (sumiyab) [inline frontend/src/components/okr/mainsSreen/More.tsx:25]: descriptive name

File: frontend/src/components/okr/mainsSreen/Popever.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+import { Popover } from "@mui/material";
+// mui Icons
+import AccessTimeOutlinedIcon from "@mui/icons-material/AccessTimeOutlined";
+import ScaleOutlinedIcon from "@mui/icons-material/ScaleOutlined";
+import DeleteOutlineOutlinedIcon from "@mui/icons-material/DeleteOutlineOutlined";
+import EditIcon from "@mui/icons-material/Edit";
+import More from "./More";
+
+const PopeverDown = props => {
+  const { handleClose, setType, setData, data, collection, setOpen, openPop, setCollection, setEdit, ind1 } = props;
```
Comment (sumiyab) [inline frontend/src/components/okr/mainsSreen/Popever.tsx:11]: descriptive name

File: frontend/src/components/okr/mainsSreen/Popever.tsx [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+      anchorOrigin={{
+        vertical: "bottom",
+        horizontal: "left",
+      }}
+    >
+      {collection.type === "dad"
+        ? ResultData.map((el, ind) => {
+            return (
+              <More
+                Icon={el.Icon}
```
Comment (sumiyab) [inline frontend/src/components/okr/mainsSreen/Popever.tsx:41]: descriptive name

File: frontend/src/components/okr/mainsSreen/Summary.tsx [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+  },
+  container: {
+    display: "flex",
+    alignItems: "center",
+    justifyContent: "space-between",
+    width: "100%",
+  },
+};
+const Summary = props => {
+  const { collection, edit, setCollection, setEdit, data, ind, setData } = props;
```
Comment (sumiyab) [inline frontend/src/components/okr/mainsSreen/Summary.tsx:36]: descriptive name

File: frontend/src/components/okr/mainsSreen/Type.tsx [hunk 7]
```diff
@@ -0,0 +1,24 @@
+import { Dispatch, SetStateAction } from "react";
+
+export type dataType = {
```
Comment (sumiyab) [inline frontend/src/components/okr/mainsSreen/Type.tsx:3]: what DataType? (descriptive name)

File: frontend/src/components/okr/mainsSreen/Type.tsx [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+  name: string;
+  child: string[];
+};
+export type PropsType = {
+  title: string;
+  setData: Dispatch<SetStateAction<unknown>>;
+  data: dataType[];
+  ind: number;
+};
+export type ColType = {
```
Comment (sumiyab) [inline frontend/src/components/okr/mainsSreen/Type.tsx:13]: descriptive name

File: frontend/src/components/okr/mainsSreen/Accordion.tsx [hunk 9]
```diff
@@ -0,0 +1,186 @@
+import { FC, SyntheticEvent, useEffect, useState } from "react";
```
Comment (sumiyab) [inline frontend/src/components/okr/mainsSreen/Accordion.tsx:1]: It is too much long lines for one file coding need to seperate two files.

**Comment by sumiyab** [comment]
svvld shinechlegdseneer ni pr description oo oorchloorei

**Review by M4nduul — APPROVED** [review]

**Pull request merged**